### ISSUE TITLE
Allow underscore in function name

### ISSUE THREAD

**Issue by mthenw** [serverless/serverless#2433]
# This is a bug report
## Description

It's pretty common to use underscore. Right now it returns error.
## Additional Data

```

  Serverless Error ---------------------------------------

     Template format error: Resource name Test_functionLambdaFunction
     is non alphanumeric.

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues

  Your Environment Infomation -----------------------------
     OS:                 darwin
     Node Version:       6.3.0
     Serverless Version: 1.0.2
```

**Comment by am17torres** [comment]
Related, though potentially not part of this bug.

I have an api gateway custom authorizer, hosted in aws, separate from my serverless project that uses underscores to separate words in the name. 

When I try to run `serverless deploy` I get the following error: 

``` bash
$ serverless deploy
Serverless: Packaging service…
Serverless: Uploading CloudFormation file to S3…
Serverless: Uploading service .zip file to S3…
Serverless: Updating Stack…

  Serverless Error ---------------------------------------

     Template format error: Resource name Jwt_custom_authorizerApiGatewayAuthorizer
     is non alphanumeric.

Stack Trace --------------------------------------------

ServerlessError: Template format error: Resource name Jwt_custom_authorizerApiGatewayAuthorizer is non alphanumeric.
  at Response.req.send (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:142:20)
  at Request.<anonymous> (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:355:18)
  at Request.callListeners (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
  at Request.emit (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
  at Request.emit (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:668:14)
  at Request.transition (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:22:10)
  at AcceptorStateMachine.runTo (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/state_machine.js:14:12)
  at ~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/state_machine.js:26:10
  at Request.<anonymous> (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:38:9)
  at Request.<anonymous> (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:670:12)
  at Request.callListeners (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/sequential_executor.js:115:18)
  at Request.emit (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
  at Request.emit (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:668:14)
  at Request.transition (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:22:10)
  at AcceptorStateMachine.runTo (~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/state_machine.js:14:12)
  at ~/.nvm/versions/node/v6.9.1/lib/node_modules/serverless/node_modules/aws-sdk/lib/state_machine.js:26:10

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues

  Your Environment Information -----------------------------
     OS:                 darwin
     Node Version:       6.9.1
     Serverless Version: 1.1.0
...
```

Here is a snippet from `serverless.yml`

```yaml
events:
      - http:
          path: api-key/create
          method: post
          authorizer: arn:aws:lambda:us-east-1:<account>:function:jwt_custom_authorizer
          cors: true
```

**Comment by eahefnawy** [comment]
Fixed with #2679 ... Closing...
